godot-core/src/builtin/string/node_path.rs

```diff
@@ -234,19 +234,13 @@ impl_rust_string_conv!(NodePath);
 
 impl From<&str> for NodePath {
     fn from(s: &str) -> Self {
-        GString::from(s).into()
-    }
-}
-
-impl From<String> for NodePath {
-    fn from(s: String) -> Self {
-        GString::from(s).into()
+        Self::from(&GString::from(s))
     }
 }
 
 impl From<&String> for NodePath {
     fn from(s: &String) -> Self {
-        GString::from(s).into()
+        Self::from(&GString::from(s))
```

Comment:
I don't like the double conversion, but I couldn't find a way in the GDExtension API to construct a `NodePath` directly from UTF-8 bytes. Maybe add a comment to clarify while you're here?